src/devices/Sht4x/Sht4x.cs

```diff
@@ -80,6 +80,25 @@ public void Reset()
             Thread.Sleep(1);
         }
 
+        /// <summary>
+        /// Reads the serial number of the device
+        /// </summary>
+        /// <returns>
+        /// A 32 bit integer with the serial number.
+        /// If a CRC check failed for a reading, it will be <see langword="null"/>.
+        /// </returns>
+        public int? ReadSerialNumber()
+        {
+            Span<byte> buffer = stackalloc byte[6];
+            _device.WriteByte((byte)0x89);
+            _device.Read(buffer);
+
+            ushort? msb = Sensirion.ReadUInt16BigEndianAndCRC8(buffer);
+            ushort? lsb = Sensirion.ReadUInt16BigEndianAndCRC8(buffer.Slice(3, 3));
+
+            return lsb + (msb << 16);
```

Comment:
why not using version instead of an int?
You definitely get 2 elements:

![image](https://github.com/dotnet/iot/assets/8145578/c49a344f-b86a-44a3-8cb7-6a12038537b7)

Reply:
In to the example C code provided by Sensirion the serial number is also output as a 32 bit integer. When the manufacturer of the chip interprets the serial number in this way I think it would be best to be consistent and also use a 32 bit integer here.

https://github.com/Sensirion/embedded-sht/blob/master/sht4x/sht4x.c#L103

Reply:
Agreed, in this code the serial is just stored in a 32 bit integer, at the pointer which was passed through at function call. There is no example code that uses it.

The Adafruit library has it also implemented in a similar way and has an example on how the serial is used:

Code:
https://github.com/adafruit/Adafruit_SHT4X/blob/master/Adafruit_SHT4x.cpp#L83

Example:
https://github.com/adafruit/Adafruit_SHT4X/blob/master/examples/SHT4test/SHT4test.ino#L26

```
Serial.println("Found SHT4x sensor");
Serial.print("Serial number 0x");
Serial.println(sht4.readSerial(), HEX);
```

Reply:
ok, still, there is no check. So I4m fine with this int.